my-app/src/App.js

```diff
@@ -1,10 +1,54 @@
 import React, { Component } from 'react';
+import ListIterator from './ListIterator';
 
 class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      items: []
+    };
+    this.userInput = React.createRef();
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleDelete = this.handleDelete.bind(this);
+  }
+
+  handleSubmit(event) {
+    let userInputElement = this.userInput.current;
+    let currentUserInput = userInputElement.value;
+    let items = [...this.state.items];
+    if (currentUserInput !== '') {
+      if (this.state.items.indexOf(currentUserInput) > -1) {
+        alert('Duplicate Input');
+      } else {
+        items.push(currentUserInput);
+        this.setState({
+          items
+        });
+      }
+      userInputElement.value = '';
+    } else {
+      alert('Enter something Jackass');
+    }
+    event.preventDefault();
```

Comment:
this should ideally be on the top/ very first line of the function. Because it prevents the default behavior of the event.